plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java

```diff
@@ -157,12 +157,9 @@ public RowType getAddEntryType(MetadataEntry metadataEntry, boolean requireWrite
     private static RowType.Field buildNullCountType(Optional<String> columnName, Type columnType)
     {
         if (columnType instanceof RowType rowType) {
-            if (columnName.isPresent()) {
-                return RowType.field(
-                        columnName.get(),
-                        RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())));
-            }
-            return RowType.field(RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())));
+            return columnName
+                    .map(name -> RowType.field(name, RowType.from(rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList()))))
```

Comment:
`rowType.getFields().stream().map(field -> buildNullCountType(field.getName(), field.getType())).collect(toImmutableList())` is common for both code branches. Let's extract it to a separate variable.